nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -0,0 +1,508 @@
+
+        if((context.getProperty(USER_NAME).isSet() && context.getProperty(PASSWORD).isSet())
+           && !context.getProperty(AUTH_BY_HEADER).isSet()){
+            final String userName = context.getProperty(USER_NAME).getValue();
+            final String password = context.getProperty(PASSWORD).getValue();
+            HttpTransportPropertiesImpl.Authenticator
+                    auth = new HttpTransportPropertiesImpl.Authenticator();
+            auth.setUsername(userName);
+            auth.setPassword(password);
+            options.setProperty(org.apache.axis2.transport.http.HTTPConstants.AUTHENTICATE, auth);
+        }
+        return options;
+    }
+
+    protected OMElement createAuthHeader(final OMFactory fac, final OMNamespace omNamespace, final OMElement parent, Map<String, Object> map) {
+        OMElement element = null;
+        if (map != null) {
+            for (String key : map.keySet()) {
+                element = fac.createOMElement(key, omNamespace);
+                Object value = map.get(key);
+                if (value instanceof String) {
+                    element.setText(value.toString());
+                    if (parent != null) {
+                        parent.addChild(element);
+                    }
+                } else {
+                    OMElement childElement = createAuthHeader(fac, omNamespace, element, (Map) value);
+                    element.addChild(childElement);
+                }
+            }
+        }
+        return element;
+    }
+
+    FlowFile processSoapRequest(ProcessSession session, final OMElement result, FlowFile flowFile, final boolean skipFirstElement) {
+        flowFile = session.write(flowFile, out -> {
+            try {
+                if (skipFirstElement) {
+                    String response = result.getFirstElement().getText();
+                    out.write(response.getBytes());
+                } else {
+                    out.write(result.toString().getBytes());
+                }
+            } catch (AxisFault axisFault) {
+                final ComponentLog logger = getLogger();
+                if (null != logger) {
+                    logger.error("Failed parsing the data that came back from the web service method", axisFault);
+                }
+                throw new ProcessException(axisFault);
+            }
+        });
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(CoreAttributes.MIME_TYPE.key(), "application/xml");
+        return session.putAllAttributes(flowFile, attributes);
+    }
+
+    OMElement executeSoapMethod(final ServiceClient serviceClient, final OMElement authHeader, final OMElement method) {
+        try {
+            if (authHeader != null) {
+                serviceClient.addHeader(authHeader);
+            }
+            getLogger().info("Sending authHeader: " + authHeader);
```

Comment:
What do you think about logging this at debug level? Otherwise it could be quite verbose when using the processor at a high rate.